framework/auth/views.py

```diff
@@ -155,9 +187,56 @@ def forgot_password_get(auth):
     return context
 
 
+@collect_auth
+def redirect_unsupported_institution(auth):
+    """
+    Sends user back to the "Unsupported Institution" page on CAS. Logs user out if they are
+    already logged in.
+
+    HTTP Method: GET
+
+    :param auth: the authentication context
+    :return
+    """
+
+    cas_unsupp_inst_url = cas.get_login_url('', campaign='unsupportedinstitution')
+
+    # if users are logged in, log them out and redirect back to this page
+    if auth.logged_in:
+        return auth_logout(redirect_url=cas_unsupp_inst_url)
+
+    return redirect(cas_unsupp_inst_url)
```

Comment:
One minor issue but not a blocker :)

It is possible that CAS is logged in and OSF is logged out. In this case, CAS will ignore the `campaign=unsupportedinstitution` query param. With `service` as an empty string `''`, this lands user on the generic login success page as shown below.

<img width="658" alt="Screen Shot 2021-02-22 at 4 44 01 PM" src="https://user-images.githubusercontent.com/3750414/108774077-36e6b700-752d-11eb-9a84-b61d845f9597.png">

Two suggestions:
* Adding OSF home as the default service URL since we expect all usages of CAS with OSF to have one.
* Option 1: force a CAS logout if OSF is already logged out (option 1)

```python
defualt_service_url = web_url_for('index', _absolute=True)  # I think this is for the OSF home.
cas_unsupp_inst_url = cas.get_login_url(defualt_service_url, campaign='unsupportedinstitution')
cas_logout_and_redirect_url = cas.get_logout_url(cas_unsupp_inst_url)

if auth.logged_in:
    return auth_logout(redirect_url=cas_unsupp_inst_url)
return redirect(cas_logout_and_redirect_url)
```

* Option 2: simply skip the `if` condition and always calls OSF `auth_logout()`.

```python
service_url = web_url_for('index', _absolute=True)  # I think this is for the OSF home.
cas_unsupp_inst_url = cas.get_login_url(service_url, campaign='unsupportedinstitution')
return auth_logout(redirect_url=cas_unsupp_inst_url)
```

Reply:
After discussion w/ @cslzchen, we've decided that this is enough of a corner case to not be a blocker for merge.  The user would need to sus the proper endpoint url to trigger this condition, while also maintaining a logged-in CAS & not logged-in OSF.  This is ...difficult, and the reward for overcoming the difficulty is a slightly worse UI.